src/main/java/Duke/Logic.java

```diff
@@ -0,0 +1,100 @@
+package Duke;
+
+import Duke.Tasks.Deadline;
+import Duke.Tasks.Event;
+import Duke.Tasks.Todo;
+
+import java.util.Scanner;
+
+public class Logic {
+    public static void loop() {
+        TaskList taskList = TextFile.loadTaskList();
+        while (true) {
+            Scanner scan = new Scanner(System.in);
+            String command = scan.next();
+            if (command.equals("bye")) {
+                break;
+            }
+            switch (command) {
+            case "list": {
+                taskList.printTaskList();
+                break;
+            }
+            case "done": {
+                int taskNumber;
```

Comment:
maybe apply Single Level of Abstraction Principle (SLAP) here? Each case command having a method for that. It would improve your code abstraction.